README.md

```diff
@@ -9,57 +9,314 @@
-  peerPublicKey: 'KRixgcBCBdyQln7IBYiopjuNO78QSFtXgOwP1sbsCSk',
-  currency: 'USD'
+  settleToBalance: async function (ctx, outgoingBalance) {
+    const claim = Network.createClaim(outgoingBalance)
+
+    return {
+      balance: outgoingBalance,
+      claim: claim
+    }
+  },
+
+  handleSettlement: async function (ctx, settlement) {
+    const { balance, claim } = settlement
+
+    if (Network.verify(claim, balance)) {
+      await ctx.state.bestClaim.setIfMax({ value: balance, data: claim })
+    }
+  },
+
+  disconnect: async function (ctx) {
+    const claim = ctx.state.bestClaim.getMax()
+    if (!claim) {
+      return
+    }
+
+    await Network.submitClaim(claim)
+  }
 })
 ```
+
+## Example Code with Unconditional Payment-Based Settlement
+
+```js
+const { MakePluginVirtual } = require('ilp-plugin-virtual')
+const { NotAcceptedError } = require('ilp-plugin-shared').Errors
+const Network = require('some-example-network')
+const BigNumber = require('bignumber.js')
+
+return MakePluginVirtual({
+  connect: async function (ctx, opts) {
+    await Network.connectToNetwork()
+
+    ctx.state.incomingSettlements = ctx.backend.getTransferLog('incoming_settlements')
+    ctx.state.amountSettled = ctx.backend.getMaxValueTracker('amount_settled')
+
+    ctx.state.maxInFlight = opts.maxInFlight
+  },
+
+  handleIncoming: async function (ctx, transfer) {
+    const incoming = await ctx.transferLog.getHighestIncomingBalance() 
+    const amountReceived = await ctx.state.incomingSettlements.getHighestIncomingBalance()
+
+    const exceeds = new BigNumber(incoming)
+      .sub(amountReceived)
+      .gt(ctx.state.maxInFlight)
+
+    if (exceeds) {
+      throw new NotAcceptedError(transfer.id + ' exceeds max in flight balance')
+    }
+  },
+
+  settleToBalance: async function (ctx, outgoingBalance) {
+    const lastPaid = ctx.state.amountSettled.setIfMax({ value: outgoingBalance, data: null })
+    const diff = outgoingBalance - lastPaid.value
+
+    if (!diff) {
+      return
+    }
+
+    const txid = Network.makePaymentToPeer(diff)
+
+    return { txid }
+  },
+
+  handleSettlement: async function (ctx, settlement) {
+    const { txid } = settlement
+    const payment = await Network.getPayment(txid)
+
+    if (!payment) {
+      return
+    }
+
+    // it doesn't really matter whether this is fulfilled or not,
+    // we just need it to affect the incoming balance so we know
+    // how much has been received. A transferlog is used here so
+    // the same txid cannot be applied twice.
+    await ctx.state.incomingSettlements.prepare({
+      id: txid,
+      amount: payment.amount
+    }, true) // isIncoming: true
+  }
+
+}
+```
+
+## Backend API, with Extensions for Payment Channels
+
+-------
+
+### `async TransferLog.getIncomingBalance ()`
+
+Get the sum of all incoming payments in the `fulfilled` state.
+
+#### Returns
+
+- `return` (Integer String) incoming balance.
+
+-------
+
+### `async TransferLog.getHighestIncomingBalance ()`
```

Comment:
(after conversation this morning) getIncomingBalance becomes getIncomingFulfilled, and getHighestIncomingBalance becomes getIncomingFulfilledAndPrepared. Same goes for outgoing.